features/LimitReuse.php

```diff
@@ -44,147 +45,107 @@ public function __construct(BetterPasswordPlugin $plugin) {
 	public function rememberPasswords($hook, $args) {
 		if ($hook == 'changepasswordform::execute' && get_class($args[0]) == 'PKP\user\form\ChangePasswordForm') {
 			$user = $args[0]->getUser();
-			$password = $args[0]->getData('oldPassword');
-			$user = $this->_addPassword($user, $password);
-			//Repo::user()->edit($user);
-			Repo::user()->edit($user);
-			//put assignment of objects here
 		}
+		else if ($hook == 'loginchangepasswordform::execute' && get_class($args[0]) == 'PKP\user\form\LoginChangePasswordForm'){
+			$user = Repo::user()->getByUsername($args[0]->getData('username'), false);
+		}
+
+		$password = $user->getData('password');
+		$this->_addPassword($user, $password);
 		return false;
 	}
 
+	/**
+	 * Checks the current form and checks if the user's inputted password has been used previously
+	 * @param string $hook The hook name
+	 * @param array $args Arguments of the hook
+	 * @return bool true if process completes
+	 */
 	public function passwordChangeValidation($hook, $args) {
 		$form = $args[0];
-		$newCheck = new FormValidatorCustom($form, 'password', 'required', 'plugins.generic.betterPassword.validation.betterPasswordPasswordReused', $this->passwordCompare($password, $form));
+		$newCheck = new FormValidatorCustom($form, 'password', 'required', 'plugins.generic.betterPassword.validation.betterPasswordPasswordReused', function ($password) use ($form) {
+			 return $this->passwordCompare($password, $form);
+		}); //$this->passwordCompare());
 		$form->addCheck($newCheck);
 		return false;
 	}
 
+	/**
+	 * Updates the time of the users last password change
+	 * @param string $password The user's given password
+	 * @param ChangePasswordForm $form The form for changing passwords
+	 * @return bool false if the user's password has been reused and true if it has not
+	 */
 	public function passwordCompare($password, $form) {
-		$user = Application::get()->getRequest()->getUser();
-		if (!$user) {
-			$user = Repo::user()->getByUsername($form->getData('username'));
+		$user = $form->_user;
+		$storedPasswordsDao = DAORegistry::getDAO('StoredPasswordsDAO');
+		if ($user) {
+			$storedPasswords = $storedPasswordsDao->getByUserId($user->getId());
+		}
+		else {
+			$storedPasswords = null;
 		}
-		foreach ($this->_getPasswords($user) as $previousPassword) {
-			if (Validation::verifyPassword($user->getUsername(), $password, $previousPassword, $rehash)) { //ask about rehash, only instance of it
-				return false;
+
+		if ($storedPasswords) {
+			if (!$user) {
+				$user = Repo::user()->getByUsername($form->getData('username'));
+			}
+			foreach ($storedPasswords->getPasswords() as $previousPassword) {
+				if (Validation::verifyPassword($user->getUsername(), $password, $previousPassword, $rehash)) {
+					return false;
+				}
 			}
 		}
 		return true;
 	}
 
 	/**
 	 * Retrieve the hashed user passwords
-	 * @param User $user
-	 * @return ?string
+	 * @param User $user The given user
+	 * @return ?string The list of passwords
```

Comment:
Is this function still used?  It looks to be based on JSON from the User settings, not our new CSV.